codecov_auth/commands/owner/interactors/set_upload_token_required.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass
+
+from codecov.commands.base import BaseInteractor
+from codecov.commands.exceptions import Unauthenticated, Unauthorized, ValidationError
+from codecov.db import sync_to_async
+from codecov_auth.helpers import current_user_part_of_org
+from codecov_auth.models import Owner
+
+
+@dataclass
+class SetUploadTokenRequiredInput:
+    upload_token_required: bool
+    org_username: str
+
+
+class SetUploadTokenRequiredInteractor(BaseInteractor):
+    def validate(self, owner_obj):
+        if not self.current_user.is_authenticated:
+            raise Unauthenticated()
+        if not owner_obj:
+            raise ValidationError("Owner not found")
+        if not current_user_part_of_org(self.current_owner, owner_obj):
+            raise Unauthorized()
+        if not owner_obj.is_admin(self.current_owner):
+            raise Unauthorized("Admin authorization required")
```

Comment:
Can upload_token_required be null? If not maybe do some validation here too

Reply:
on it